### DISCUSSION TITLE
Expose credentials to the Vite process

### DISCUSSION THREAD

**Discussion by pepicrft** [ElMassimo/vite_ruby#96] [Q&A] [answered]
Hi 👋🏼 ,
I was wondering if it's possible to expose Rails' credentials to the Vite subprocess. For example, with Webpacker the following API is available to achieve so:

```ruby
Webpacker::Compiler.env['VALUE'] = Rails.application.credentials.dig(:value)
```

**Comment by ElMassimo** [comment, marked as answer]
Hi Pedro! The equivalent would be:

```ruby
ViteRuby.env['VALUE'] = Rails.application.credentials.dig(:value)
```

If you are planning to use the environment variable in client code, you might want to check [`vite-plugin-environment`](https://github.com/ElMassimo/vite-plugin-environment), or alternatively [prefix it with `VITE_`](https://vitejs.dev/guide/env-and-mode.html).

**Comment by vitalyliber** [comment]
@ElMassimo I'm sorry, but I didn't understand where I should put this?
`ViteRuby.env['VALUE'] = Rails.application.credentials.dig(:value)`

I'd tried to add it to `config/initializers/vite.rb`, but I can't get the value in the client.
```
import.meta.env.VALUE
// undefined
import.meta.env.VITE_ADDRESS_FINDER
// undefined
```

**Reply by ElMassimo** [reply]
Hi @vitalyliber!

The best place to configure env variables for Vite is in [`config/vite.rb`](https://vite-ruby.netlify.app/config/#ruby-configuration-file-%F0%9F%92%8E), an optional [_Ruby Configuration File_](https://vite-ruby.netlify.app/config/#ruby-configuration-file-%F0%9F%92%8E).

Also, if you want to access it from client code, make sure to [allow those env variables](https://vitejs.dev/config/shared-options.html#envprefix), or name them with a `VITE_` prefix, which is allowed by default:

```ruby
# config/vite.rb
ViteRuby.env['VITE_VALUE'] = Rails.application.credentials.dig(:value)
```

**Reply by Iwaide** [reply]
Excuse me for jumping in. It seems that Rails.application is not available in config/vite.rb,
so we might need to add the following lines to either config/vite.rb or bin/vite:

```
require_relative '../config/application'
```

Please let me know if adding this could cause any issues. Thank you!

**Reply by ElMassimo** [reply]
Normally, it is available, as it's either loaded by the [`environment`](https://github.com/ElMassimo/vite_ruby/blob/main/vite_rails/lib/tasks/vite.rake#L5) rake task precondition or [explicitly](https://github.com/ElMassimo/vite_ruby/blob/main/vite_rails/lib/vite_rails/cli.rb#L16) when running `vite build`.

Requiring it for `vite dev` would make the server startup slower, and is not a good trade-off, as most apps don't use `config/vite.rb`, and those that do can require it manually.

**Reply by Iwaide** [reply]
You're right, I was only considering the development environment. I'll go ahead and manually write the require statement. Thank you for pointing that out.
